src/querier.js

```diff
@@ -1,101 +1,98 @@
-		  "useQueryLabels": "true",
-		  "dateTimeUtc": "true",
-		  "queries": [
-		    {
-		      "name": "Subscriptions",
-		      "query": subsQuery ,
-		      "type": "zoqlexport"
-		    }
-		    ,
-		    {
-		      "name": "HolidaySuspensions",
-		      "query": holidaySuspensionQuery,
-		      "type": "zoqlexport"
-		    }
-		  ]
-		  },                
-		  headers: {               
-		    'Authorization': 'Basic ' + new Buffer(`${config.username}:${config.password}`).toString('base64'),
-		    'Content-Type': 'application/json'                  
-		  }
-		};  
+    const options = {
+      method: 'POST',
+      uri: `${config.url}/apps/api/batch-query/`,
+      json: true,
+      body: {
+        'format': 'csv',
+        'version': '1.0',
+        'name': 'Fulfilment-Queries',
+        'encrypted': 'none',
+        'useQueryLabels': 'true',
+        'dateTimeUtc': 'true',
+        'queries': [
+          {
+            'name': 'Subscriptions',
+            'query': subsQuery,
+            'type': 'zoqlexport'
+          },
+          {
+            'name': 'HolidaySuspensions',
+            'query': holidaySuspensionQuery,
+            'type': 'zoqlexport'
+          }
+        ]
+      },
+      headers: {
+        'Authorization': 'Basic ' + Buffer.from(`${config.username}:${config.password}`).toString('base64'),
+        'Content-Type': 'application/json'
+      }
+    }
 
-		request(options, function(error, response, body) {  
-		if (error) {
-			reject(error);
-			return;
-			}	
+    request(options, function (error, response, body) {
+      if (error) {
+        reject(error)
+        return
+      }
 
-		  console.log('statusCode:', response && response.statusCode); 
-		  console.log('body:', body); 
-		  if(response.statusCode != '200') {
-		  		reject(`error response status ${response.statusCode}`);
-			} else if(body.errorCode) {
-				reject(`zuora error! code: ${body.errorCode} : ${body.message}`)
-			} else {
-				resolve(body.id);
-			}
-		});
-	});
-	return promise;
+      console.log('statusCode:', response && response.statusCode)
+      console.log('body:', body)
+      if (response.statusCode !== '200') {
+        reject(new Error(`error response status ${response.statusCode}`))
+      } else if (body.errorCode) {
+        reject(new Error(`zuora error! code: ${body.errorCode} : ${body.message}`))
+      } else {
+        resolve(body.id)
+      }
+    })
+  })
+  return promise
 }
 
-
-
-export function handler(input, context, callback){
-	fetchConfig()
-	.then(config=> queryZuora(input.deliveryDate, config))
-	.then(jobId => callback(null, {"jobId":jobId, "deliveryDate":input.deliveryDate}))
-	.catch(error => callback(error));
-}
\ No newline at end of file
+export function handler (input, context, callback) {
+  fetchConfig()
+  .then(config => queryZuora(input.deliveryDate, config))
+  .then(jobId => callback(null, {'jobId': jobId, 'deliveryDate': input.deliveryDate}))
+  .catch(error => callback(error))
```

Comment:
I'm not sure why but running it locally I don't get the full error description here unless I do callback(error.toString()) but that probably will not be interpreted correctly by the state machine